src/lib/core/tests/TestTLV.cpp

```diff
@@ -2836,6 +2838,60 @@ void CheckTLVCharSpan(nlTestSuite * inSuite, void * inContext)
     }
 }
 
+void CheckTLVGetLocalizedStringIdentifier(nlTestSuite * inSuite, void * inContext)
+{
+    struct CharSpanTestCase
+    {
+        const char * testString;
+        Optional<LocalizedStringIdentifier> expectedLSID;
+    };
+
+    // clang-format off
+    static CharSpanTestCase sCharSpanTestCases[] = {
+        // Test String                                Expected LocalizedStringIdentifier from Get()
+        // ==============================================================================================
+        {  "This is a test case #0",                  chip::Optional<LocalizedStringIdentifier>()       },
+        {  "This is a test case #1\0370123",          chip::Optional<LocalizedStringIdentifier>(0x0123) },
```

Comment:
Suggest `\x1F` instead of `\037` and suggest maing a macro `#define IS1_CHAR "\x1F"` so you can do `"This is a test" IS1_CHAR "0123"`, etc

Reply:
updated